kivy/__init__.py

```diff
@@ -381,44 +381,47 @@ def _patch_mod_deps_win(dep_mod, mod_name):
     Logger.setLevel(level=level)
 
     # Can be overridden in command line
-    if ('KIVY_UNITTEST' not in environ and
-            'KIVY_PACKAGING' not in environ and
-            environ.get('KIVY_NO_ARGS', "false") not in ('true', '1', 'yes')):
-        # save sys argv, otherwise, gstreamer use it and display help..
-        sys_argv = sys.argv
-        sys.argv = sys.argv[:1]
+    try:
+        if ('KIVY_UNITTEST' not in environ and
+                'KIVY_PACKAGING' not in environ and
+                environ.get('KIVY_NO_ARGS', "false") not in ('true', '1', 'yes')):
+            # save sys argv, otherwise, gstreamer use it and display help..
+            sys_argv = sys.argv
+            sys.argv = sys.argv[:1]
 
-        try:
-            opts, args = getopt(sys_argv[1:], 'hp:fkawFem:sr:dc:', [
-                'help', 'fullscreen', 'windowed', 'fps', 'event',
-                'module=', 'save', 'fake-fullscreen', 'auto-fullscreen',
-                'multiprocessing-fork', 'display=', 'size=', 'rotate=',
-                'config=', 'debug', 'dpi='])
-
-        except GetoptError as err:
-            Logger.error('Core: %s' % str(err))
-            kivy_usage()
-            sys.exit(2)
+            try:
+                opts, args = getopt(sys_argv[1:], 'hp:fkawFem:sr:dc:', [
+                    'help', 'fullscreen', 'windowed', 'fps', 'event',
+                    'module=', 'save', 'fake-fullscreen', 'auto-fullscreen',
+                    'multiprocessing-fork', 'display=', 'size=', 'rotate=',
+                    'config=', 'debug', 'dpi='])
+
+            except GetoptError as err:
+                Logger.error('Core: %s' % str(err))
+                kivy_usage()
+                sys.exit(2)
+
+            mp_fork = None
+            try:
+                for opt, arg in opts:
+                    if opt == '--multiprocessing-fork':
+                        mp_fork = True
+                        break
+            except:
+                pass
 
-        mp_fork = None
-        try:
-            for opt, arg in opts:
-                if opt == '--multiprocessing-fork':
-                    mp_fork = True
-                    break
-        except:
-            pass
-
-        # set argv to the non-read args
-        sys.argv = sys_argv[0:1] + args
-        if mp_fork is not None:
-            # Needs to be first opt for support_freeze to work
-            sys.argv.insert(1, '--multiprocessing-fork')
+            # set argv to the non-read args
+            sys.argv = sys_argv[0:1] + args
+            if mp_fork is not None:
+                # Needs to be first opt for support_freeze to work
+                sys.argv.insert(1, '--multiprocessing-fork')
 
-    else:
+        else:
+            opts = []
+            args = []
+    except:
```

Comment:
This is for worse than failing, as it skips over the actual error. Without any warning /indication either.